### PR TITLE
RD-267  Remove jQuery-UI (#1473)

### PR DESCRIPTION
**Pull request by stepek**
* RD-267 Remove jQuery and jQuery-UI

### CODE REVIEW

**Review by Gelio — COMMENTED** [review]
What's the purpose of this PR? I see that there is #1492 and #1473 and I can't figure out what's happening

**Comment by stepek** [comment]
This should be merged after removing jQuery from repos.

**Comment by qooban** [comment]
This is to be merged only after https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/162 is merged and new version of cloudify-blueprint-topology is released, right?

**Comment by stepek** [comment]
@qooban Yes, before that we should merger #162

**Comment by qooban** [comment]
I assume you had https://github.com/cloudify-cosmo/cloudify-blueprint-topology/pull/162 in mind :-) 
OK, so let's finalize PR in cloudify-blueprint-topology repo first, then release new version and update the version here. Then we can review it finally.

**Comment by stepek** [comment]
@qooban `cloudify-blueprint-topology` updated

File: package.json [hunk 1]
```diff
@@ -58,7 +58,6 @@
     "i18next": "^19.8.3",
     "isomorphic-fetch": "^3.0.0",
     "jquery": "^3.5.0",
```
Comment (Gelio) [inline package.json:60]: Can we also remove `jquery`, similarly to https://github.com/cloudify-cosmo/cloudify-blueprint-composer/pull/905#discussion_r674550763?

File: webpack.config.js [hunk 2]
```diff
@@ -160,7 +160,6 @@ module.exports = (env, argv) => {
             resolve: {
                 extensions: resolveExtensions,
                 alias: {
-                    'jquery-ui': 'jquery-ui/ui',
                     jquery: `${__dirname}/node_modules/jquery`, // Always make sure we take jquery from the same place
```
Comment (Gelio) [inline webpack.config.js:163]: If you will remove jquery, let's also remember to remove this line

Comment (Gelio) [reply webpack.config.js:163]: There are also some uses of `jquery` down below in `webpack.config.js`

**Review by Gelio — COMMENTED** [review]
btw. I see that jquery is also used in `test/jest/setup.ts`. Maybe we could remove it from there too?

File: package.json (hunk 1 above)
Comment (stepek) [reply package.json:60]: No, i'll do this after removing last jquery usage. https://github.com/cloudify-cosmo/cloudify-stage/pull/1491

Comment (Gelio) [reply package.json:60]: So what's the order of merging the PRs :smile: ? There are 3 open PRs for stage regarding jQuery and I've lost track

Comment (qooban) [reply package.json:60]: I've lost track, too. I think we should discuss that on UI weekly (I'll add a point). Looks like there are (or at least there were) many open PRs related to jQuery removal at the same time which are not independent and all of us are engaged in it. That's not optimal situation I think.

**Review by qooban — COMMENTED** [review]
I see that there will be more changes in this PR as it looks like the CI is failing due to size of the package, so I'll postpone my review until it is finalized.

Can you check the size? That's probably because the Topology widget bundle is now bigger (contains jQuery) and we still have jQuery in the deps? 

That's, IMHO, another reason for having merging order of all of jQuery removal PRs settled and clarified.

**Comment by stepek** [comment]
### System Tests 
https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/1001/

File: package-lock.json [hunk 3]
```diff
@@ -21809,11 +21816,6 @@
         "logalot": "^2.0.0"
       }
     },
-    "jquery": {
-      "version": "3.5.0",
-      "resolved": "https://registry.npmjs.org/jquery/-/jquery-3.5.0.tgz",
-      "integrity": "sha512-Xb7SVYMvygPxbFMpTFQiHh1J7HClEaThguL15N/Gg37Lri/qKyhRGZYzHRyLH8Stq3Aow0LsHO2O2ci86fCrNQ=="
-    },
     "jquery-ui": {
```
Comment (Gelio) [inline package-lock.json:21819]: Minor:

Interesting... `jquery-ui` is still there, and `jquery` is not (it's moved to be a direct dependency of `cloudify-blueprint-topology`). Can you check why there is such a difference between the 2 packages?

I thought we could remove it entirely and keep it only included in `cloudify-blueprint-topology` :thinking:

**Review by Gelio — APPROVED** [review]
Looks clean :+1: Nice work getting rid of jquery :smile: 

I see that `jquery-ui` is still at the top-level of `package-lock.json`, which worries me :thinking: Let's check why the way it's stored is different from how `jquery` is stored (it's not on the top level, but it only is present as a dependency of `cloudify-blueprint-topology`)

**Review by qooban — APPROVED** [review]
Looks good, just double check `package-lock.json` file as mentioned by @Gelio .

File: package-lock.json (hunk 3 above)
Comment (stepek) [reply package-lock.json]: Fixed,

Comment (Gelio) [reply package-lock.json]: What did you to do fix it? What was the problem :smile: ? Did you move it manually?

**Comment by stepek** [comment]
https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/1008/ pass

**Review by qooban — APPROVED** [review]

**Pull request merged**